### ISSUE TITLE
Path to ammeter device not found

### ISSUE THREAD

**Issue by k0pernicus** [JonHylands/fxos-powertool#15]
Hi!

I try to use Powertool for my experiments but it seems i canno't found the path to the ammeter device.
The ammeter device is a YoctoAmp model, and I'm working on a ElementaryOS desktop.

This is the log from `dmesg`:

```
[13786.612940] usb 1-5: new full-speed USB device number 18 using xhci_hcd
[13786.743073] usb 1-5: New USB device found, idVendor=24e0, idProduct=001f
[13786.743075] usb 1-5: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[13786.743077] usb 1-5: Product: Yocto-Amp
[13786.743077] usb 1-5: Manufacturer: Yoctopuce
[13786.743078] usb 1-5: SerialNumber: YAMPMK01-20866
[13786.744588] hid-generic 0003:24E0:001F.0011: hiddev0,hidraw2: USB HID v1.11 Device [Yoctopuce Yocto-Amp] on usb-0000:00:14.0-5/input0
```

I try to see if there is a serial port opened in /dev for this device, but it seems not.  
Do you have an idea about to find this path please?

Thanks a lot!

**Comment by JonHylands** [comment]
You need to set up a udev rule for this device.

**Comment by k0pernicus** [comment]
Thanks!

I wrote a rule for this device named `81-yoctoampdrive.rules`, based on the informations given by the command `udevadm info -a -p $(udevadm info -q path -n /dev/usb/hiddev0)`:

```
KERNEL=="hidraw[0-5]", SUBSYSTEMS=="usb" ,ATTRS{idVendor}=="XXXX", ATTRS{idProduct}=="XXXX", ATTRS{serial}=="XXXX", SYMLINK+="ammeter"
```

I test it and it append to `/dev` a new mount point named `ammeter`.

But, when I want to test the program on this path (`sudo powertool -d yocto -p /dev/ammeter -u tk -s current -o some_tests.csv`), Powertool said to me : 

```
Exception:
 from /usr/local/lib/python2.7/dist-packages/powertool-0.3-py2.7.egg/powertool/sample_source.py, line 49:
 Failed to load: mozilla (REASON: SerialException -- Could not configure port: (22, 'Invalid argument'))
```

Is the rule failed?

**Comment by k0pernicus** [comment]
Based on this [Stackoverflow post](http://stackoverflow.com/questions/28691680/can-pyserial-work-with-hid-devices), it seems that Pyserial doesn't work with `hidraw` devices... :-/
Instead of writing my own USB driver, do you already have this problem before?

Thanks!

**Comment by JonHylands** [comment]
I'm not sure exactly what you're trying to do here. I don't know why you're trying to do this "hiddev/hidraw" stuff - I've never heard of that.

Just make a simple udev rule, like this:

```
ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="204b", ENV{ID_MM_DEVICE_IGNORE}="1"
ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="204b", ENV{MTP_NO_PROBE}="1"
SUBSYSTEM=="tty", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="204b", MODE="0666", GROUP="dialout"
```

replace the vendor id and product id with the values you get from lsusb. You should end up with a device at /dev/ttyACM0

**Comment by JonHylands** [comment]
Alternately, you could use their provided udev rule:

https://www.yoctopuce.com/EN/products/virtualhub/doc/VIRTHUB0.usermanual.html

**Comment by k0pernicus** [comment]
Ok, thanks a lot.

I tried your solution with my values:

```
Bus 001 Device 004: ID 24e0:001f 
```

with this rule (in `/etc/udev/rules.d/51-yoctopuce.rules`):

```
ATTRS{idVendor}=="24e0", ATTR{idProduct}=="001f", ENV{ID_MM_DEVICE_IGNORE}="1"
ATTRS{idVendor}=="24e0", ATTR{idProduct}=="001f", ENV{MTP_NO_PROBE}="1"
SUBSYSTEM=="tty", ATTR{idVendor}=="24e0", ATTR{idProduct}=="001f", MODE="0666"
```

I reboot my computer but nothing, I don't have any `/dev/ttyUSBX` or `/dev/ttyACMX`...  
VirtualHub works, and I tried to where the device has been "mounted" in /dev, but no way to find this information...

**Comment by JonHylands** [comment]
Well, until you can get /dev/ttyXXXX with that device, I don't think I can help much. Maybe try and talk to Yoctopuce, and see what they say. The above works for me in Ubuntu. I don't know anything about ElementaryOS...

**Comment by k0pernicus** [comment]
Ok, thanks for the help :-)

I just forgot to make the rule file executable, but nothing change... :-/

I will try on Ubuntu 15.10 before to talk to Yoctopuce...

Thanks again!

**Comment by k0pernicus** [comment]
I just tried on a Ubuntu 15.10 and it doesn't works too...
Just before to add the rule, the device was recognized in `/dev/usb/hiddev0` and `/dev/hidraw2`.
Since the rule has been added and the computer rebooted, there is no `/dev/usb` and no `/dev/hidraw2` - also, no `ttyACMX` or `ttyUSBX`. Just some `tty0` to `tty63` and some `ttyS0` to `ttyS31`.

I'll send an email to Yoctopuce explaining the situation.
